cloudimg/ms_azure.py

```diff
@@ -375,6 +376,15 @@ def upload_to_container(self, image_path, container_name, object_name,
             create=True
         )
 
+        # Azure can't handle compressed images on marketplaces so we need to
+        # send the decompressed data to its storage account.
+        if object_name.endswith(".xz"):
+            log.info("Processing a LZMA compressed file.")
+            object_name = object_name.rstrip(".xz")
```

Comment:
It would be nice to document  rstrip behaviour and automatic decompression 
at least in the publish method

Reply:
tyvm